### DISCUSSION TITLE
Disabling markdown diagnostic when entering a buffer

### DISCUSSION THREAD

**Discussion by pidgeon777** [LazyVim/LazyVim#5185] [Q&A]
Actually, when entering a markdown buffer, some diagnostic is shown (reporting long lines, etc.).

I would prefer not to show diagnostic for a markdown buffer, when I enter it.

I know that there is the LazyVim "<leader>ud" keybinding for toggling diagnostic.

There are two main problems with it:

1. This will toggle diagnostic for ALL buffers.
2. While diagnostic can be toggled to *disabled* status, a function such as the following:

```lua
--- Get diagnostics in a given range
--- @param bufnr number
--- @param start_line number
--- @param end_line number
--- @return table<CopilotChat.select.selection.diagnostic>|nil
local function get_diagnostics_in_range(bufnr, start_line, end_line)
  local diagnostics = vim.diagnostic.get(bufnr)
  local range_diagnostics = {}
  local severity = {
    [1] = 'ERROR',
    [2] = 'WARNING',
    [3] = 'INFORMATION',
    [4] = 'HINT',
  }

  for _, diagnostic in ipairs(diagnostics) do
    local lnum = diagnostic.lnum + 1
    if lnum >= start_line and lnum <= end_line then
      table.insert(range_diagnostics, {
        severity = severity[diagnostic.severity],
        content = diagnostic.message,
        start_line = lnum,
        end_line = diagnostic.end_lnum and diagnostic.end_lnum + 1 or lnum,
      })
    end
  end

  return #range_diagnostics > 0 and range_diagnostics or nil
end
```

will still return the selection diagnostic in any case.

So, my question is: would it be possible someway to toggle the diagnostic for a buffer only (or maybe even better for a given filetype), completely disabling the diagnostic (so that a function such as the above one would return nil)?

**Comment by dpetka2001** [comment]
`:h vim.diagnostic.enable()`

You can pass the number of buffer to it like `:lua vim.diagnostic.enable(false, { bufnr = 0 })` (0 for current buffer). But enabling diagnostics globally in some other buffer with `<leader>ud` will still enable the diagnostics you disabled for the buffer. Maybe bind the above command to a mapping of your preference and use that whenever you want. 

For filetypes, I think you would have to create an autocommand on `FileType` event and check with `pattern` for the filetype you want the autocommand to be enabled and in the `callback` just put the above command. Haven't tested it, but I believe it should work. But the minor disadvantage with this is that if you later enable globally diagnostics again, I believe they would also be enabled for the buffer of the filetype as well and the autocommand would not trigger again because the `FileType` event would have already occured once. So, maybe a custom mapping would be better? 

Test it out and see which approach fits best your needs.
